bolt/CMakeLists.txt

```diff
@@ -135,36 +134,168 @@ if (LLVM_INCLUDE_TESTS)
   endif()
 endif()
 
-if (BOLT_ENABLE_RUNTIME)
-  message(STATUS "Building BOLT runtime libraries for X86")
-  set(extra_args "")
-  if(CMAKE_SYSROOT)
-    list(APPEND extra_args -DCMAKE_SYSROOT=${CMAKE_SYSROOT})
+set(AARCH64_GNU_C_COMPILER aarch64-linux-gnu-gcc)
+set(AARCH64_GNU_CXX_COMPILER aarch64-linux-gnu-g++)
+set(X86_64_GNU_C_COMPILER x86_64-linux-gnu-gcc)
+set(X86_64_GNU_CXX_COMPILER x86_64-linux-gnu-g++)
+set(RISCV_GNU_C_COMPILER riscv64-linux-gnu-gcc)
+set(RISCV_GNU_CXX_COMPILER riscv64-linux-gnu-g++)
```

Comment:
* Should these be cacheable `option()` or [`set( ... CACHE ...)`](https://cmake.org/cmake/help/latest/command/set.html#set-cache-entry) so that they can be set by the user? (What if the user wants to point it at a specific clang which is configured for the target?)
* Should they be namespaced by starting with `BOLT_`?
* Perhaps you can drop GNU from the name, I'm not sure it adds much.

Reply:
The point with GNU in the name is, that these are only used in case the compiler inferred from the CC and CXX envvars is a GNU compiler, and not a generic option.

Reply:
It still seems conceivable someone could use a different non-gnu compiler though? Or they might want to point this to some other compiler which is not the one which happens to be used to compile BOLT.

Reply:
[As you can see here](https://github.com/bgergely0/llvm-project/blob/main/bolt/CMakeLists.txt#L241-L270) I only check for Clang and GNU compilers, and not something else. So the user would still not be able to compile with an arbitrary compiler, without adding support here.

What else should be supported? Is the limitation to only Clang/GNU not acceptable?

Reply:
My request is that the user should be able to specify the cross compiler to use.

Essentially, what if the user wants to do `-DBOLT_AARCH64_CXX_COMPILER=foobar`?. I think this should be fine, and it's not GNU. Ideally it should 'win' over the auto-detection as well.

That said, the user could probably also invoke cmake manually just for the runtimes for this use case, so maybe what you have - which I understand is opportunistically attempting to auto-detect if we can reuse the compiler used to compile bolt - is fine.